### PR TITLE
Lightswitch fixes + roundstart area light up refactor

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Fixes light switches that start off at roundstart turning their lights on instantly and silently when they're activated for the first time.
Completely changes the logic of how lights are lit/extinguished based on departments being staffed - now all lights initially start on, and when the ticker sends out the COMSIG_GLOB_POST_START signal, lights in areas that aren't staffed get turned off.
Removes the "in area" for loops from light switch code and replaces them with in GLOB.machinery for optimization.

## Why It's Good For The Game
Bugfix good, less processing time also good

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/110184118/e9e95fc1-b9cf-48d9-b6c2-da10d850ccd2

</details>

## Changelog
:cl:
fix: Fixes light switches that start turned off due to an empty department, turning their lights on instantly and silently for the first time
code: Refactored the process for making un-staffed departments be dark at roundstart
code: Slightly reduced the processing cost of flicking light switches
/:cl:

### CODE REVIEW

**Review by Rukofamicom — APPROVED** [review]

**Comment by itsmeow** [comment]
> Removes the "in area" for loops from light switch code

We have cheaper methods of getting `in area` now, since areas track their turfs. It might actually be cheaper to use that now. (Added in #9469)

**Comment by PowerfulBacon** [comment]
Its not cheaper to use that

File: code/game/machinery/lightswitch.dm [hunk 1]
```diff
@@ -49,13 +40,21 @@
...
+		return
+	if(length(GLOB.roundstart_areas_lights_on))
+		if(area in GLOB.roundstart_areas_lights_on)//Department is staffed, lights should shart on
+			return
+	area.lightswitch = FALSE //All checks failed, department is not staffed, lights get turned off
+
+	for(var/obj/machinery/light_switch/L in GLOB.machines)
+		if(L.area == area)
+			L.update_appearance(updates = UPDATE_ICON|UPDATE_OVERLAYS)
+	message_admins("Turning off lights in [area]")
```
Comment (EvilDragonfiend) [inline code/game/machinery/lightswitch.dm:57]: eh leftover admin message?

Comment (CydiaLamiales) [reply code/game/machinery/lightswitch.dm:57]: Ah fuck, yep

**Pull request merged**